src/pages/api/revalidate-all-tutorials.ts

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import type { NextApiRequest, NextApiResponse } from 'next/types'
+import { StatusCodes } from 'http-status-codes'
+
+import { validateToken } from 'lib/api-validate-token'
+import { getAllCollections } from 'lib/learn-client/api/collection'
+import {
+	Collection as ClientCollection,
+	TutorialLite as ClientTutorialLite,
+} from 'lib/learn-client/types'
+import { activeProductSlugs } from 'lib/products'
+import { SectionOption } from 'lib/learn-client/types'
+import { getTutorialSlug } from 'views/collection-view/helpers'
+import { getCollectionSlug } from 'views/collection-view/helpers'
+import { ProductSlug } from 'types/products'
+
+const BATCH_SIZE = 5
+
+/**
+ * Accepts a POST request, triggers revalidation for all tutorial paths for all products
+ * landing pages, collection pages, and tutorial pages associated
+ */
+async function handler(request: NextApiRequest, response: NextApiResponse) {
+	if (request.method !== 'POST') {
+		response.status(StatusCodes.NOT_FOUND)
+		return
+	}
+
+	try {
+		const tutorialLandingPaths = getTutorialLandingPaths()
+		const collectionAndTutorialPaths = await getCollectionAndTutorialPaths()
+		const paths: string[] = [
+			...tutorialLandingPaths,
+			...collectionAndTutorialPaths,
+		]
+
+		if (!paths || paths.length === 0) {
+			response.status(200).end()
+		}
+
+		// Sending 'accepted' status to avoid socket hang-up as the path revaliadation takes a very long time
+		response
+			.status(StatusCodes.ACCEPTED)
+			.send(`Revalidating all tutorials: ${paths.length} paths`)
```

Comment:
@BRKalow would this be a better response since we're expecting this batch to take a really long time and the client could timeout? I experienced a socket hangup when testing with postman locally and wondered if we should keep the response open while waiting for all of these to settle.

Reply:
@kendallstrautman I think that's reasonable yeah. The potential risk here is that we don't have visibility into the status of the revalidation, but we can iterate on that. Any idea how long the full revalidation takes?

Reply:
Yeah... I'm unsure the right way to handle this as an API newb. Locally I've seen it take between 5-8 minutes.

Reply:
Okay yeah looks like we get a [socket hangup in the action](https://github.com/hashicorp/tutorials/actions/runs/4325350508/jobs/7551391150) if we wait for everything to resolve. 
<img width="1572" alt="Screenshot 2023-03-03 at 8 35 10 AM" src="https://user-images.githubusercontent.com/36613477/222775950-0ad80e48-f5a2-48ce-ba84-ed9719cbefc9.png">

Reply:
😱